src/memory/memoryManager.ts

```diff
@@ -20,54 +19,28 @@ export class MemorySegmentManager {
     return ptr;
   }
 
-  loadData(
-    address: Relocatable,
-    data: MaybeRelocatable[]
-  ): Result<Relocatable> {
+  setData(address: Relocatable, data: MaybeRelocatable[]): Relocatable {
     for (const [index, d] of data.entries()) {
-      const { value: indexUint, error: indexError } =
-        UnsignedInteger.toUint32(index);
-      if (indexError !== undefined) {
-        return { value: undefined, error: indexError };
-      }
-      const { value: nextAddress, error: addrError } = address.add(indexUint);
-      if (addrError !== undefined) {
-        return { value: undefined, error: addrError };
-      }
+      const nextAddress = address.add(index);
       this.insert_inner(nextAddress, d);
     }
 
-    const { value: dataLen, error: dataLenError } = UnsignedInteger.toUint32(
-      data.length
-    );
-    if (dataLenError !== undefined) {
-      return { value: undefined, error: dataLenError };
-    }
+    const newSegmentSize =
+      this.getSegmentSize(address.getSegmentIndex()) + data.length;
+    UnsignedInteger.ensureUint32(newSegmentSize);
 
-    const { value: newSegmentSize, error: segmentSizeError } =
-      UnsignedInteger.toUint32(
-        this.getSegmentSize(address.getSegmentIndex()) + dataLen
-      );
-    if (segmentSizeError !== undefined) {
-      return { value: undefined, error: segmentSizeError };
-    }
     this.segmentSizes.set(address.getSegmentIndex(), newSegmentSize);
 
-    return address.add(dataLen);
+    return address.add(data.length);
   }
 
   // Insert a value in the memory at the given address and increase
   // the segment size by 1.
   insert(address: Relocatable, value: MaybeRelocatable): void {
     this.insert_inner(address, value);
     if (address.getOffset() > this.getSegmentSize(address.getSegmentIndex())) {
-      const { value: newSize, error } = UnsignedInteger.toUint32(
-        address.getOffset() + UnsignedInteger.ONE_UINT32
-      );
-      if (error !== undefined) {
-        throw error;
-      }
-      this.segmentSizes.set(address.getSegmentIndex(), newSize);
+      UnsignedInteger.ensureUint32(address.getOffset() + 1);
```

Comment:
do we really need this? a segment could be just an array and in js, it's possible to do

```
a[0] = 1
a[10] = 1
```

and `a.length` would return the `segmentSize`

Reply:
actually i think this makes sense, in the effort of making everything as simple as possible